dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -88,12 +100,18 @@ public class IconController {
   @GetMapping("/{iconKey}/icon.svg")
   public void getIconData(HttpServletResponse response, @PathVariable String iconKey)
       throws IOException, NotFoundException {
-    Resource icon = iconService.getDefaultIconResource(iconKey);
+    downloadDefaultIcon(iconKey, response);
+  }
 
-    response.setHeader("Cache-Control", CacheControl.maxAge(TTL, TimeUnit.DAYS).getHeaderValue());
-    response.setContentType(MediaType.SVG_UTF_8.toString());
+  @GetMapping(value = "/{uid}/icon")
+  public void getCustomIconData(@PathVariable String uid, HttpServletResponse response)
+      throws NotFoundException, WebMessageException, IOException {
 
-    StreamUtils.copyThenCloseInputStream(icon.getInputStream(), response.getOutputStream());
+    if (iconService.iconExists(uid)) {
```

Comment:
This code is confusing to me 😅 How is it that if an `iconExists` it is a default icon and otherwise a custom icon? This is not obvious from an `iconService` that to me is returning `icons` 🤷🏻 Maybe @muilpp can clarify

Even more confusing is that
* the method is called `getCustomIconData` but also returns default icons.
* and `getIconData` only `downloadDefaultIcon` with a path of `"/{iconKey}/icon.svg"`. How does that path convey that to a user? When the path `"/{uid}/icon"` returns default and custom icons 😅

Reply:
default icons are bundled icons identified by a key. The requirement is to have a single endpoint for both 
custom and default icons. So if the key exists it means it's a default icon otherwise find it in the custom icon list.
As far as `getCustomIconData` name is concerned it can be changed `getCustomOrDefaultIConData`

Reply:
@zubaira, if I understand this piece of code correctly, you expect `iconService.iconExists(uid)` to return true only if a default icon exists with that given uid, but [the implementation of this method](https://github.com/dhis2/dhis2-core/blob/DHIS2-15409/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultIconService.java#L134) checks both default icons and custom icons, meaning that if any of these two kind of icons has that uid, it will return true.

Reply:
good catch @muilpp. Fixed it